Add styles for User Pages

RayRedGoose opened the pull request:
### This PR contains the next changes:

- Added User Card to user profile page. 
- Added Edit account page for managing user information 
- Added a 404 image for unavailable edit  page

Also, I added new route:
- `u/:username/edit` for edit account page

and I added a `vue-scrollactive`  package for scrolling following menu in an edit page

@wlonk This PR is in progress because I need to redesign a user page layout.

wlonk commented:
Oh, also the textbox for user bio edit can be stretched or shrunk horizontally, and that super breaks the layout.

RayRedGoose commented:
> Oh, also the textbox for user bio edit can be stretched or shrunk horizontally, and that super breaks the layout.

I fixed that, thanks for a catch!

RayRedGoose commented:
> I like how this is factored and organized, and I really like the scrollspy. I'm gonna do some work on it to add functionality and change a few minor things to names I prefer, but I think it's really good.

Let me finish implementing designs for user page and polishing it before you start to do it :)

> I looked over the mock-ups in figma, and I think the tabbed view (groups across the top, relevant games full width under them) is good, as we're unlikely to have a user be more than part of, say, five groups.

I added components to a user profile with a group tab panel and user's games block. I added pagination for tabs to show how it can work (it's not working yet, I'll add this functionality later). Now it looks like completely filled profile, but I need to do some cleaning and fixing small bugs.

> I also think that the notifs preferences needs a change. I realized that weekly in-app is weird and makes no sense. A more likely setup is that for each kind of notif, we have three options, as checkboxes: in-app, instant email, weekly email.

Done. I have to variation of it: as a row and a column.
<img width="506" alt="Screen Shot 2020-08-23 at 10 07 37 PM" src="https://user-images.githubusercontent.com/48605821/91003188-e8f8f280-e58d-11ea-970a-c942824265ef.png">
<img width="471" alt="Screen Shot 2020-08-23 at 10 13 34 PM" src="https://user-images.githubusercontent.com/48605821/91003190-e9918900-e58d-11ea-927d-191baa35879c.png">

RayRedGoose commented:
I would like to receive any other feedback from you to improve a visual representation.

wlonk commented on src/components/UserGames.vue:
```diff
@@ -0,0 +1,39 @@
+        team: { name: "Transneptune Games" },
+        latest_version: { name: "v0.2.2" },
+        permissions: { delete: true, },
+      }]
```
Just a reminder to clean this up to pull data from the API, and clean up the template to not list the games 3x.

wlonk commented on src/views/UserDetail.vue:
```diff
@@ -10,11 +15,36 @@
+        'Best group',
+        'more group',
+        'One one more group',
+        'one Best group',
```
Again, reminder to eventually pull this from the API.

RayRedGoose commented:
`this.$on(...)` in `created` is used for creating an event to pass as props to the child and invoke a function changed parent state after a click.

wlonk commented on src/views/UserDetail.vue:
```diff
@@ -20,6 +54,7 @@ export default {
   },
   created() {
     this.$store.dispatch('getUserById', this.$route.params.username);
+    this.$on('chooseGroup');
```
https://vuejs.org/v2/guide/components-custom-events.html#Event-Names recommends that you always use kebab-case for custom event names. I wonder if we should follow that so far as to make all custom events have at least two words, to avoid accidental collision with pre-existing events?

wlonk commented on src/views/EditUserPage.vue:
```diff
@@ -0,0 +1,45 @@
+        </div>
+      </div>
+    </div>
+    <NoPageNotification v-else />
```
I'm gonna abstract this out into a generic 404 handler later; it hasn't been a priority yet though.

wlonk commented:
Do you still wanna do some design on this, or can I dive in to the API integration work? Thanks!

RayRedGoose commented:
> Do you still wanna do some design on this, or can I dive in to the API integration work? Thanks!

Since you have no comment about the design changes, so I don't plan to add anything to this PR. The only thing that I plan to add to these pages is dark/white mode switch, but It will be another PR. I've started designing main pages and I'm going to work on it.

wlonk commented:
Wonderful, thanks. I'll add some API changes in to this and get it merged forthwith!
